next/src/components/_global/nav/nav.tsx

```diff
@@ -0,0 +1,174 @@
+        scrollDistance += prevScrollPos - currentScrollPos;
+        if (scrollDistance * -1 >= navHeight) {
+          nav.classList.remove(`${styles.fixed}`, 'fixed');
+          scrollDistance = 0;
+        }
+      }
+      if (currentScrollPos === 0) {
+        nav.classList.remove(styles.fixed);
+      }
+    };
+    window.addEventListener('scroll', handleScroll);
+    window.addEventListener('resize', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+      window.removeEventListener('resize', handleScroll);
+    };
+  }, []);
+
+  return (
+    <nav
+      className={styles.navWrapper}
+      ref={navRef}
+    >
+      <div className={`maxWidth ${styles.nav}`}>
+        <Img
+          className={styles.logo}
+          data={logo}
+        />
+        <div className={styles.linksWrapper}>
+          <div className={styles.links}>
+            <Link href={'/o-mnie'}>O mnie</Link>
+            <div className={isExpanded ? `${styles.dropdown} ${styles.expanded}` : styles.dropdown}>
+              <button
+                onClick={() => setIsExpanded(!isExpanded)}
+                className={styles.button}
+              >
+                <span>Oferta</span>
+                {ChevronDown()}
+              </button>
+              {isExpanded && (
+                <>
+                  <Link href={'/co-robie'}>Co robię</Link>
+                  <Link href={'/efekty-wspolpracy'}>Efekty współpracy</Link>
+                </>
+              )}
+            </div>
+            <Link href={'/Blog'}>Blog</Link>
+          </div>
+        </div>
+        <Button
+          className={styles.button}
+          svg={false}
+          data={cta}
+        />
+        <div className={styles.hamburger}>{Hamburger({ toggleNav, setToggleNav })}</div>
+      </div>
+      {toggleNav && (
+        <div className={styles.toggled}>
+          <Img
+            className={styles.logo}
+            data={logo}
+          />
+          {Exit({ toggleNav, setToggleNav })}
+          <div className={styles.links}>
+            <Link href={'/o-mnie'}>O mnie</Link>
+            <Link href={'/Oferta'}>Oferta</Link>
+            <div className={styles.sublinks}>
+              <Link href={'/Co-robie'}>Co robię</Link>
+              <Link href={'/Efekty-wspolpracy'}>Efekty współpracy</Link>
+            </div>
+            <Link href={'/Blog'}>Blog</Link>
+          </div>
+          <Button
+            className={styles.button}
+            svg={false}
+            data={cta}
+          />
+        </div>
+      )}
+    </nav>
+  );
+}
+
+export function Hamburger({
+  toggleNav,
+  setToggleNav,
+}: {
+  toggleNav: boolean;
+  setToggleNav: (toggleNav: boolean) => void;
+}) {
+  return (
+    <svg
+      xmlns='http://www.w3.org/2000/svg'
+      width='62'
+      height='61'
+      fill='none'
+      viewBox='0 0 62 61'
+      onClick={() => setToggleNav(!toggleNav)}
```

Comment:
`onClick` on SVG? Accessibility, you should use button with `onClick` and inside of that svg element